server/server.go

```diff
@@ -74,14 +76,41 @@ func checkServerRunning(bind string) bool {
 	return err == nil
 }
 
+func StartCredentialProxyOnWindows() error {
+	log.Printf("Starting `aws-vault server` in the background")
+	cmd := exec.Command(os.Args[0], "server")
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+	time.Sleep(time.Second * 1)
+	if !checkServerRunning(metadataBind) {
+		return errors.New("The credential proxy server isn't running. Run aws-vault server as Administrator in the background and then try this command again")
+	}
+	return nil
+}
+
+func StartCredentialProxyWithSudo() error {
+	log.Printf("Starting `aws-vault server` as root in the background")
+	cmd := exec.Command("sudo", "-b", os.Args[0], "server")
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func StartCredentialProxy() error {
+	if runtime.GOOS == "windows" {
+		return StartCredentialProxyOnWindows()
+	}
+	return StartCredentialProxyWithSudo()
```

Comment:
@williamdenton Actually another way to do this is put the arch specific functions into conditional build files, rather than using the runtime check. What's preferred @lox ?

Reply:
It gets complicated, let's stick to this switch. LGTM.